qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -1,18 +1,330 @@
+                           initial_value=0.0,
+                           unit='V',
+                           vals=vals.Numbers(-25, 25),
+                           parameter_class=GroupParameter,
+                           docstring=textwrap.dedent("""
+        Start value of the stair case sweep (in V). For the log sweep, 
+        start and stop must have the same polarity.
+                                """))
+
+        self.add_parameter(name='sweep_end',
+                           initial_value=0.0,
+                           unit='V',
+                           vals=vals.Numbers(-25, 25),
+                           parameter_class=GroupParameter,
+                           docstring=textwrap.dedent("""
+        Stop value of the DC bias sweep (in V). For the log sweep,start and
+        stop must have the same polarity.
+                                """))
+
+        self.add_parameter(name='sweep_steps',
+                           initial_value=1,
+                           vals=vals.Ints(1, 1001),
+                           parameter_class=GroupParameter,
+                           docstring=textwrap.dedent("""
+        Number of steps for staircase sweep. Possible  values from 1 to 
+        1001"""))
+
+        self.add_parameter(name='current_compliance',
+                           initial_value=100e-3,
+                           unit='A',
+                           vals=vals.Numbers(-40, 40),
+                           parameter_class=GroupParameter,
+                           docstring=textwrap.dedent("""
+        Current compliance (in A). Refer to Manual 2016. See Table 4-7 on 
+        page 24, Table 4-9 on page 26, Table 4-12 on page 27, or Table 4-15 
+        on page 28 for each measurement resource type. If you do not set 
+        current_compliance, the previous value is used.
+        Compliance polarity is automatically set to the same polarity as the
+        output value, regardless of the specified Icomp. 
+        If the output value is 0, the compliance polarity is positive. If 
+        you set Pcomp, the maximum Icomp value for the measurement resource 
+        is allowed, regardless of the output range setting.
+                           """))
+
+        self.add_parameter(name='power_compliance',
+                           initial_value=2.0,
+                           unit='W',
+                           parameter_class=GroupParameter,
+                           vals=vals.Numbers(0.001, 80),
+                           docstring=textwrap.dedent("""
+        Power compliance (in W). Resolution: 0.001 W. If it is not entered, 
+        the power compliance is not set. This parameter is not available for
+        HVSMU. 0.001 to 2 for MPSMU/HRSMU, 0.001 to 20 for HPSMU, 0.001 to 
+        40 for HCSMU, 0.001 to 80 for dual HCSMU, 0.001 to 3 for MCSMU, 
+        0.001 to 100 for UHVU
+                           """))
+
+        self.add_parameter(name='chan',
```

Comment:
i wouldn't want users to change this parameter, so we should make it private with `_chan` name. also for CMU occurences.

Reply:
done